mcodingbot/database/models/user.py

```diff
@@ -18,3 +24,65 @@ async def get_or_create(user_id: int) -> User:
             return await User(user_id=user_id).create()
         except UniqueViolationError:
             return await User.fetch(user_id=user_id)
+
+    @staticmethod
+    async def add_word(word: str, user_id: int) -> None:
+        word_model = await Word.get_or_create(word=word)
+        user = await User.get_or_create(user_id=user_id)
+        await word_model.users.add(user)
+
+    @staticmethod
+    async def delete_word(word: str, user_id: int) -> bool:
+        """
+        Return `True` if the word was successfully removed from the user.
+        Return `False` otherwise.
+        """
+        if not await User.exists(user_id=user_id):
+            return False
+
+        user = await User.fetch(user_id=user_id)
+        words = await user.words.fetchmany()
+
+        for word_ in words:
+            if word_.word == word:
+                break
+        else:
+            return False
+
+        try:
+            word_model = await Word.fetch(word=word)
+        except ModelNotFound:
+            return False
+
+        await user.words.remove(word_model)
+        return True
+
+
+@final
```

Comment:
I don't think we need this.

Reply:
the typehints in the static method are technically wrong without it

Reply:
how so?

Reply:
```python
class A:
    @staticmethod
     def f() -> A"
 
class B(A):
    ...
```
B now return A when you call the static method and thats not expected. So just say you can't subclass at all because you aren't supposed to subclass A.

Reply:
Ill remove the final decorator